src/pages/ResearcherProfile.js

```diff
@@ -1,492 +1,351 @@
-      prev.profile.eRACommonsID = '';
-      return prev;
-    });
-  };
+  const handleRadioChange = (event, field, value) => {
+    let newFields = {
+      isThePI: profile.isThePI,
+      havePI: profile.havePI
+    };
 
-  clearPIData = () => {
-    this.setState(prev => {
-      prev.profile.piName = '';
-      prev.profile.piEmail = '';
-      return prev;
-    });
-  };
+    newFields[field] = value;
 
-  saveProfile = (event) => {
-    event.preventDefault();
-    this.setState({ showDialogSave: true });
-  };
+    if (profile.isThePI === true || profile.isThePI === 'true') {
+      newFields.havePI = '';
+    }
+    if (profile.havePI === false || profile.havePI === 'false' || profile.havePI === '') {
+      newFields.piName = '';
+      newFields.piEmail = '';
+      newFields.piERACommonsID = '';
+    }
 
-  cleanObject = (obj) => {
-    // Removes any zero length properties from a copy of the object
-    return omitBy(obj, (s) => { return isNil(s) || trim(s.toString()).length === 0; });
+    setProfile(Object.assign({}, profile, newFields));
   };
 
-  saveProperties = async (profile) => {
-    await Researcher.createProperties(profile);
-    await this.saveUser();
-    this.setState({ showDialogSubmit: false });
-    this.props.history.push({ pathname: 'dataset_catalog' });
+  const eraValidate = async () => {
+    const user = await User.getMe();
+    const userProps = getPropertyValuesFromUser(user);
+    const expirationCount = isNil(userProps.eraExpiration) ? 0 : AuthenticateNIH.expirationCount(userProps.eraExpiration);
+
+    return (!isNil(userProps.eraCommonsId) && userProps.eraAuthorized === 'true' && expirationCount >= 0);
+
+    // Temporary fix until eRACommons.js is updated to share re-render info.
   };
 
-  dialogHandlerSubmit = (answer) => async () => {
-    if (answer === true) {
-      let profile = this.state.profile;
-      profile = this.cleanObject(profile);
-      profile.completed = true;
-      if (this.state.profile.completed === undefined) {
-        await this.saveProperties(profile);
-      } else {
-        await this.updateResearcher(profile);
-      }
+  const submitForm = async (event) => {
+    event.preventDefault();
 
+    const eraValid = await eraValidate();
+    const profileCompleted = researcherFieldsComplete && eraValid;
+
+    const newProfile = Object.assign({}, profile, {completed: profileCompleted});
+
+    if (isNewProfile) {
+      await createResearcher(newProfile);
     } else {
-      this.setState({ showDialogSubmit: false });
+      await updateResearcher(newProfile);
     }
   };
 
-  updateResearcher = async (profile) => {
-    const profileClone = this.cloneProfile(profile);
-    await Researcher.updateProperties(Storage.getCurrentUser().dacUserId, true, profileClone);
-    await this.saveUser();
-    this.setState({ showDialogSubmit: false });
-    this.props.history.push({ pathname: 'dataset_catalog' });
+  const createResearcher = async (profile) => {
```

Comment:
Let's update this method name to reflect what it's really doing. Naming is hard!  I think it should be:
```suggestion
  const createUserProperties = async (profile) => {
```
Additionally, and separately, `Researcher.createProperties(profile);` should be renamed/moved to `User.createProperties(profile);`

Reply:
To be clear, 
> Additionally, and separately, Researcher.createProperties(profile); should be renamed/moved to User.createProperties(profile);

is meant for a separate PR, not this one.

Reply:
Fixed, will post issue to Jira after next push